File: src/dnd/items/Club.java
package dnd.items;

import dnd.combat.DamageDiceGroup;
import dnd.data.DamageType;
import dnd.data.WeaponProperty;

/**
 * A class representing the Club item
 * 
 * @author calvi
 *
 */
public class Club extends Item {

	/**
	 * Constructor for class Club
	 */
	public Club() {
		super("Club", new double[] { 20.0, 40.0 }, 5.0);
		weaponProperties.add(WeaponProperty.LIGHT);
	}

	@Override
	public DamageDiceGroup getDamageDice() {
		return new DamageDiceGroup(1, 4, DamageType.BLUDGEONING);
	}

}

File: scripts/items/club.lua
function define()
	DamageType = luajava.bindClass("com.dndsuite.dnd.data.DamageType")
	Item = luajava.bindClass("com.dndsuite.core.Item")
	ItemAttack = luajava.bindClass("com.dndsuite.core.events.ItemAttack")
	
	self:setName("Club")
	self:addTag(Item.LIGHT)
	self:addTag(Item.SIMPLE_MELEE)
	self:addTag(Item.WEAPON)
	self:addTag(Item.CLUB)
end



function acAbilityBonusLimit()
	return 0
end



function acbase()
	return 0
end



function customTasks()
end



function damage()
	if attackType == ItemAttack.MELEE then
		return luajava.newInstance("com.dndsuite.dnd.combat.DamageDiceGroup", 1, 4, DamageType.BLUDGEONING)
	elseif attackType == ItemAttack.RANGED then
		return nil
	elseif attackType == ItemAttack.THROWN then
		return luajava.newInstance("com.dndsuite.dnd.combat.DamageDiceGroup", 1, 4, DamageType.BLUDGEONING)
	end
end



function equip()
end



function range()
	return 20.0, 40.0, -1.0, -1.0
end



function reach()
	return 0.0
end



function unequip()
end



function weight()
	return 2.0
end